Allow to configure cgroupsv1 per nodepool (#348)

fiunchinho opened the pull request:
* Allow to configure cgroupsv1 per nodepool

* Apply suggestions from code review



* Regenerate docs

* Move flatcar cgroupsv1 function to worker helper

* Adjust tests to new containerd cgroups configuration

* Apply suggestions from code review



* Still support old field

---------

### What does this PR do?

(Please set a descriptive PR title. Use this space for additional explanations.)

### What is the effect of this change to users?

### How does it look like?

(Please add anything that represents the change visually. Screenshots, output, logs, ...)

### Any background context you can provide?

(Please link public issues or summarize if not public.)

### What is needed from the reviewers?

### Do the docs need to be updated?

### Should this change be mentioned in the release notes?

- [ ] CHANGELOG.md has been updated (if it exists)

Gacko commented:
I changed the PR's title as there is, IMHO, no need to prefix it this way. Apart from that: LGTM! I guess you only cherry-picked, right?

I guess it's currently hard to impossible to test stuff like this by just running cluster test suites, so we might need to come up with a test release that uses the test version of `cluster-aws` that includes this version of `cluster` here.

fiunchinho commented:
> I changed the PR's title as there is, IMHO, no need to prefix it this way. Apart from that: LGTM! I guess you only cherry-picked, right?
> 
> I guess it's currently hard to impossible to test stuff like this by just running cluster test suites, so we might need to come up with a test release that uses the test version of `cluster-aws` that includes this version of `cluster` here.

Yeah, basically cherry-picked and fixed conflicts.
